src/movies.js

```diff
@@ -1,34 +1,112 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
-
-
+//console.log('movies: ', movies);
+const data = [
+  {
+    title: 'The Shawshank Redemption',
+    year: 1994,
+    director: 'Frank Darabont',
+    duration: '2h 22min',
+    genre: ['Crime', 'Drama'],
+    score: 9.3
+  },
+  {
+    title: 'The Godfather',
+    year: 1994,
+    director: 'Francis Ford Coppola',
+    duration: '2h',
+    genre: ['Crime', 'Drama'],
+    score: 9.2
+  },
+  {
+    title: 'The Godfather: Part II',
+    year: 1974,
+    director: 'Francis Ford Coppola',
+    duration: '3h 22min',
+    genre: ['Crime', 'Drama'],
+    score: 9
+  },
+  {
+    title: 'The Dark Knight',
+    year: 2008,
+    director: 'Steven Spielberg',
+    duration: '2h 32min',
+    genre: ['Action', 'Crime', 'Thriller']
+  }
+];
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
-
+function getAllDirectors(arr) {
+  let newArr = arr.map((el) => el.director);
+  return [...new Set(newArr)];
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
-
+function howManyMovies(arr) {
+  let newArr = arr.filter((el) => {
+    return (
+      el.director === 'Steven Spielberg' &&
+      el.genre.some((value) => value === 'Drama')
```

Comment:
The logic is very correct but you could also do:
`el.genre.includes("Drama")`
for a shorter code